git_hg_sync/retry.py

```diff
@@ -0,0 +1,34 @@
+import time
+from collections.abc import Callable
+
+from mozlog import get_proxy_logger
+
+logger = get_proxy_logger("retry")
+
+
+def retry(
+    action: str,
+    callback: Callable,
+    *,
+    tries: int = 2,
+    delay: float = 0.25,
+) -> None:
+    logger.debug(action)
+    for attempt in range(1, tries + 1):
+        try:
+            callback()
+            break
+        except Exception as exc:
```

Comment:
Is there a risk of some exception being caught here that would indicate a "partial" success, where we risk repeating certain actions in `callback`?

Reply:
It depends on what the callback does. I think there is a risk, but the callbacks we call should be idempotent.